src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -28,7 +28,7 @@ public CollectionTallyResult(List<object> missingItems, List<object> extraItems)
             }
         }
 
-        private readonly NUnitEqualityComparer comparer;
+        private readonly NUnitEqualityComparer _comparer;
```

Comment:
Not technically a pattern matching update, but this makes this class-level variable naming consistent with below (and removes a compiler INFO message in the process)